.github/workflows/build.yml

```diff
@@ -0,0 +1,121 @@
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+        pg_major: [ 11, 10 ]
+        include:
+          - pg_major: 11
+            edb_version: '11.4-2'
+            postgis_version: '2.5.3'
+          - pg_major: 10
+            edb_version: '10.9-2'
+            postgis_version: '2.5.3'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Start PostgreSQL ${{ matrix.pg_major }} (Linux)
+        if: startsWith(matrix.os, 'ubuntu')
+        run: docker run --detach --name postgresql -p 5432:5432 npgsql/postgres:${{ matrix.pg_major }}
+        shell: bash
+
+      - name: Start PostgreSQL ${{ matrix.pg_major }} (Windows)
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          # Install PostgreSQL
+          curl -o pgsql.zip -L https://get.enterprisedb.com/postgresql/postgresql-${{ matrix.edb_version }}-windows-x64-binaries.zip
+          unzip pgsql.zip -x 'pgsql/include/**' 'pgsql/doc/**' 'pgsql/pgAdmin 4/**' 'pgsql/StackBuilder/**'
+
+          # Match Npgsql CI Docker image and stash one level up
+          cp {$GITHUB_WORKSPACE/.build/docker,pgsql}/server.crt
+          cp {$GITHUB_WORKSPACE/.build/docker,pgsql}/server.key
+
+          # Install PostGIS
+          POSTGIS_FILE="postgis-bundle-pg${{ matrix.pg_major }}-${{ matrix.postgis_version }}x64"
+          curl -o postgis.zip -L https://download.osgeo.org/postgis/windows/pg${{ matrix.pg_major }}/${POSTGIS_FILE}.zip
+          unzip postgis.zip -d postgis
+          cp -a postgis/$POSTGIS_FILE/. pgsql/
+
+          # Start PostgreSQL
+          pgsql/bin/initdb -D pgsql/PGDATA -E UTF8 -U postgres
+          pgsql/bin/pg_ctl -D pgsql/PGDATA -l logfile -o '-c max_prepared_transactions=10 -c ssl=true -c ssl_cert_file=../server.crt -c ssl_key_file=../server.key' start
+
+          # Configure test account
+          pgsql/bin/psql -U postgres -c "CREATE ROLE npgsql_tests SUPERUSER LOGIN PASSWORD 'npgsql_tests'"
+          pgsql/bin/psql -U postgres -c "CREATE DATABASE npgsql_tests OWNER npgsql_tests"
+        shell: bash
+
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v1.0.2 # Issues with DOTNET_ROOT
+        with:
+          dotnet-version: '3.0.100'
+
+      - name: Test
+        run: dotnet test --configuration Debug
+        shell: bash
+
+  publish:
+    needs: build
+
+    runs-on: windows-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Pack NuGet packages (CI versions)
+        if: startsWith(github.ref, 'refs/heads/hotfix/') == false
+        run: |
+          VERSION_SUFFIX="ci.$(date -u +%Y%m%dT%H%M%S)+sha.${GITHUB_SHA:0:9}"
+          dotnet pack --configuration Release --output nupkgs --version-suffix $VERSION_SUFFIX
+        shell: bash
+
+      - name: Pack NuGet packages (Release versions)
+        if: startsWith(github.ref, 'refs/heads/hotfix/') == true
```

Comment:
This isn't quite correct - a push to hotfix/* is still a CI build, not a release... The way this currently is, all pushes to hotfix branches will produce the same nuget version (because there's no version suffix).

Here is how thinks currently work at appveyor:

* Any push to hotfix/*. This is a patch CI build - so with CI naming (need a running version suffix). nuget should get uploaded to the myget patch feed (AKA stable feed).
* Any push to dev or other branch. This is a vnext CI build, so with CI naming. nuget should get uploaded to the myget vnext feed (AKA unstable feed).
* Any push with a version *tag* (e.g. v1.2.3). This is the only scenario without a version suffix. At this point we don't need to upload this to any nuget feed - I upload it manually to nuget.org. In the future we may decide to automate this too (scary!).

The above isn't set in stone or anything - we can definitely change the flow if it makes sense.

Reply:
Okay, how does this look?

`github.ref`                                | CI suffix | gpr | vnext | patch | nuget |
:-------------------------------------------|-----------|-----|-------|-------|-------|
`refs/heads/*`, excl. `refs/heads/hotfix/*` | ✔         | ✔  | ✔    | ❌    | ❌    |
`refs/heads/hotfix/*`                       | ✔         | ✔  | ❌   | ✔     | ❌    |
`refs/tags/v*`                              | ❌        | ❌ | ❌   | ❌    | ✔     |

where

feed    | url                                                         |
--------|-------------------------------------------------------------|
`gpr`   | `https://nuget.pkg.github.com/npgsql/index.json`            |
`vnext` | `https://www.myget.org/F/npgsql-unstable/api/v3/index.json` |
`patch` | `https://www.myget.org/F/npgsql/api/v3/index.json`          |
`nuget` | `https://api.nuget.org/v3/index.json`                       |

Reply:
Okay, how does this look?

`github.ref`                                | CI suffix | gpr | vnext | patch | nuget |
:-------------------------------------------|-----------|-----|-------|-------|-------|
`refs/heads/*`, excl. `refs/heads/hotfix/*` | ✔         | ✔  | ✔    | ❌    | ❌    |
`refs/heads/hotfix/*`                       | ✔         | ✔  | ❌   | ✔     | ❌    |
`refs/tags/v*`                              | ❌        | ❌ | ❌   | ❌    | ✔     |

where

feed    | url                                                         |
--------|-------------------------------------------------------------|
`gpr`   | `https://nuget.pkg.github.com/npgsql/index.json`            |
`vnext` | `https://www.myget.org/F/npgsql-unstable/api/v3/index.json` |
`patch` | `https://www.myget.org/F/npgsql/api/v3/index.json`          |
`nuget` | `https://api.nuget.org/v3/index.json`                       |